packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,510 @@
+    LOG_ERROR("fail to set display mode :%s", get_error_message(ret));
+    return false;
+  }
+  return true;
+}
+
+int64_t VideoPlayer::Create() {
+  int ret;
+  player_id_ = gPlayerIndex++;
+  if (uri_.empty()) {
+    LOG_ERROR("uri is empty");
+    return -1;
+  }
+
+  if (!Open(uri_)) {
+    LOG_ERROR("open failed, uri : %s", uri_.c_str());
+    return -1;
+  }
+
+  if (!SetDisplay(registrar_ref_)) {
+    LOG_ERROR("fail to set display");
+    return -1;
+  }
+  SetDisplayRoi(0, 0, 1, 1);
+
+  ret = player_set_buffering_cb(player_, OnBuffering, (void *)this);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_buffering_cb failed: %s",
+              get_error_message(ret));
+  }
+
+  ret = player_set_completed_cb(player_, OnPlayCompleted, (void *)this);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_completed_cb failed: %s",
+              get_error_message(ret));
+  }
+
+  ret = player_set_interrupted_cb(player_, onInterrupted, (void *)this);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_interrupted_cb failed: %s",
+              get_error_message(ret));
+  }
+
+  ret = player_set_display_visible(player_, true);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_display_visible failed: %s",
+              get_error_message(ret));
+  }
+
+  ret = player_set_error_cb(player_, OnError, (void *)this);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_error_cb failed: %s",
+              get_error_message(ret));
+  }
+
+  ret = player_prepare_async(player_, OnPrepared, (void *)this);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_prepare_async failed: %s",
+              get_error_message(ret));
+  }
+
+  flutter::PluginRegistrar *plugin_registrar =
+      flutter::PluginRegistrarManager::GetInstance()
+          ->GetRegistrar<flutter::PluginRegistrar>(registrar_ref_);
+  SetupEventChannel(plugin_registrar->messenger());
+  return player_id_;
+}
+
+void VideoPlayer::SetDisplayRoi(int x, int y, int w, int h) {
+  int ret = player_set_display_roi_area(player_, x, y, w, h);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("fail to set display roi :%s", get_error_message(ret));
+  }
+}
+
+VideoPlayer::~VideoPlayer() {
+  if (drm_manager_) {
+    drm_manager_->ReleaseDrmSession();
+  }
+  Dispose();
+}
+
+void VideoPlayer::Play() {
+  player_state_e state;
+  int ret = player_get_state(player_, &state);
+  if (state < PLAYER_STATE_READY) {
+    player_destroy(player_);
+    LOG_ERROR("invalid state for play operation");
+    return;
+  }
+  ret = player_start(player_);
+  if (state == PLAYER_STATE_READY || state == PLAYER_STATE_PAUSED) {
+    if (ret != PLAYER_ERROR_NONE) {
```

Comment:
These `if condition` can be combined using `&&`.

Reply:
> These `if condition` can be combined using `&&`.

have fixed